base/client.jl

```diff
@@ -169,24 +169,50 @@ function _parse_input_line_core(s::String, filename::String)
     return ex
 end
 
-function parse_input_line(s::String; filename::String="none", depwarn=true)
+"""
+    parse_input_line(s::AbstractString; filename::AbstractString = "none", depwarn::Bool = true)
+
+Parse the contents of `s` as Julia code.
+If `filename` is provided, it will be used as the filename
+in all `LineNumberNode`s.
+If `depwarn` is `false`, deprecation warnings will be suppressed.
+
+```julia
+julia> expr = parse_input_line("c = sqrt(a^2 + b^2)")
+```
+
+```julia
+julia> filename = "hello.jl"
+
+julia> expr = parse_input_line(read(filename, String); filename = filename)
+```
+"""
+function parse_input_line(s::String; filename::AbstractString = "none", depwarn::Bool = true)
```

Comment:
I suspect we can remove the `depwarn` for now; as far as I can tell, the parser doesn't depwarn anymore.

Reply:
In the new `Meta.parsefile` function, I removed the `depwarn` keyword argument.